assert.d.ts

```diff
@@ -0,0 +1,68 @@
+export type AssertPredicate =
+  | RegExp
+  | (new () => object)
+  | ((thrown: unknown) => boolean)
+  | {
+      readonly name?: string | RegExp
+      readonly message?: string | RegExp
+      readonly [k: string]: unknown
+    }
+  | Error
+
+export interface StrictAssertModule {
```

Comment:
The only bit I'm a little worried about is having the `assert.d.ts` define the assertion module that comes directly from Node. There likely won't be huge differences between Node versions, but there's a possibility for a type mismatch, right? 🤔 Though, I suppose that since there aren't any official types, this is probably the most straightforward way to go.

The only way I can see to bypass that would be to have the user inject their own assertion implementation, but that'd be a breaking change and likely a little too paranoid.

Reply:
@bearror
Yes, there's a possibility for a type mismatch. However, the assert module seems pretty stable.

Another possibility is to add a dependency or a peer-dependency to @types/node. However I see several disadvantages:

1. In the case of a peer-dependency, the user needs to install itself the dependency
2. The types declared in @types/node for the assert module use type-assertions. This imposes the declaration of the type of 't' in any TS files when using most of the functions provided by assert.

One advantage to maintain our own type declaration is to control what is exposed. I forget to mention in the PR message that I deliberately exposed a subset of the strict assert module. The following exports are missing:

- AssertionError is not exported
- the module is not the assert function (impose the use of `t.ok()` instead of `t()`

We could add other restrictions, but I guess it is more controversial and need more discussions / through.